content/languages/c/authoring/index.md

```diff
@@ -0,0 +1,285 @@
+- Some header files include other header files indirectly, for example, file `foo.h` contains line `#include <bar.h>`, which might appear to make the explicit include for `bar.h` unnecessary. It's not true though, because the file `foo.h` might change one day, or might depend on some compiler settings or command line options, and after some changes to the configuration of the C runner, the `bar.h` might be not included there anymore. That's why every file (i.e. code snippet) of a kata should explicitly include all required header files declaring functions used in it.
+- Author might think that header files for the testing framework are included automatically by the code runner. It is not the case though, and test suites need to include `criterion/criterion.h` explicitly.
+
+### Compilation warnings
+
+Compiler options related to warnings used by C runner are somewhat strict and require some discipline to get the code to compile cleanly. `-Wall` and `-Wextra` cause that warnings can be numerous, and some of them are very pedantic. However, code of C kata should still compile cleanly, without any warnings logged to the console. Even when a warning does not cause any problem with tests, users get distracted by them and blame them for failing tests.
+
+
+### Memory management
+
+Unlike many modern, high-level languages, C does not manage memory automatically. Manual memory management is a very vast and complex topic, with many possible ways of achieving the goal depending on a specific case, caveats, and pitfalls.
+
+Whenever a kata needs to return a string or an array, C authors tend to use the naive technique of allocating the memory in the solution function, and freeing it in the test suite. This approach mimics the behavior known from other languages where returning an array or object from inside of user solution is perfectly valid, but it's hardly ever a valid way of working with unmanaged memory.
+
+One of the consequences of non-managed memory is that it's strongly recommended to avoid requirement of returning string constants from C solution, especially when translating kata from other languages. Returning a string in other languages is not a problem, but in C it always rises questions of who and how should allocate it. Consider replacing the string with some simpler data type (eventually aliased with a `typedef`), and/or provide some symbolic constants for available values. For exmple, if for Python version the requirement would be to _Return string `'You won!'` if you won the game, `'You lost :('` if you were defeated, and `'It's a draw'` if there is no winner._, C should preferably use constants of `WON`, `LOST` and `DRAW`. If author decides to keep raw C-strings as elements of the kata interface, they should clearly specify required alloction scheme.
```

Comment:
```suggestion
One of the consequences of unmanaged memory is that it's strongly recommended against returning string constants from C functions, especially when translating kata from other languages. Returning a string in other languages is not a problem, but in C it always raises questions of who should allocate it and how it should be allocated. Consider replacing the string with some simpler data type (eventually aliased with a `typedef`), and/or provide some symbolic constants for available values. For example, if the requirement for the Python version is to _Return the string `'You won!'` if you won the game, `'You lost :('` if you were defeated, and `'It's a draw'` if there is no winner._, C should preferably use the constants `WON`, `LOST` and `DRAW` defined in an `enum`. If the author decides to keep raw C-strings as elements of the kata interface, they should clearly specify the required allocation scheme.
```

Reply:
I've deliberately not included the _"defined in an `enum`"_ part here, and leave authors more freedom on this subject. It's true that an enum would be the best thing style-wise, but in a kata, if they want, they can use `int`, `char`, or even `const char *const` with full message content defined in preloaded, as long as it does not leak to the user to make them worry about memory allocation.

I will remove _"defined in an `enum`"_ part and merge the wording suggestions, but if you think it must stay, then let me know.